src/typ/environment.rs

```diff
@@ -127,12 +140,13 @@ impl<'a, 'b> Environment<'a, 'b> {
         name: String,
         variant: ValueConstructorVariant,
         typ: Arc<Type>,
+        origin: SrcSpan,
     ) {
         let _ = self.local_values.insert(
             name,
             ValueConstructor {
                 public: false,
-                origin: Default::default(), // TODO: use the real one
```

Comment:
The random location lines here and there are for removing this `TODO`. The initial implementation I went for used this function as an entry point so I had to have the correct origin. The current implementation does not use this function but I kept the changes related to that `TODO`. Happy to take them out if it crowds this PR or if another solution is desired.

Reply:
Sounds good to me!